adbms/src/main/scala/de/up/hpi/informationsystems/adbms/protocols/DefaultMessagingProtocol.scala

```diff
@@ -11,10 +12,31 @@ object DefaultMessagingProtocol {
     * Use this message to directly insert data into the relations of a `Dactor` implementing
     * [[de.up.hpi.informationsystems.adbms.protocols.DefaultMessageHandling]].
     * The `Dactor`s will return with a message from [[akka.actor.Status]].
+    *
     * @note Use with caution! This message relies on internal details of `Dactor`s and could lead to tight coupling.
     * @param relation name of the relation in regards
     * @param records to be inserted records
     */
   case class InsertIntoRelation(relation: String, records: Seq[Record])
 
+  /**
+    * Use this message to request the contents of a relation from a `Dactor` implementing
+    * [[de.up.hpi.informationsystems.adbms.protocols.DefaultMessageHandling]].
+    * The `Dactor` will return a
+    * [[de.up.hpi.informationsystems.adbms.protocols.DefaultMessagingProtocol.RelationQuerySuccess]]
+    * message in case of success, or a [[akka.actor.Status.Failure]] in case of failure.
+    *
+    * @note Use with caution! This message relies on internal details of `Dactor`s and could lead to tight coupling.
+    * @param relation name of the requested relation
+    */
+  case class RelationQuery(relation: String)
```

Comment:
Should I put this into an object, i.e. do you think it would reduce clutter?

I could make this:
```scala
object RelationQuery {
  case class Request(relation: String)
  case class Success(relation: Relation)
}
```

The respective messages would then be created as
```scala
DefaultMessagingProtocol.RelationQuery.Request("some_relation")
```

I relied on `akka.actor.Status.Failure` for the failure case but let me know if this should then also be a special case class.

Reply:
I like having a way to match on a specific response. I would also incorporate the failure-case:
```scala
object RelationQuery {
  case class Request(relation: String)
  case class Success(relation: Relation)
  case class Failure(e: Throwable)
}
```
Otherwise you cannot match the response to your request, see this example:
```scala
// some client actor:
[...]
val userDactor = Dactor.dactorSelection(...)
userDactor ! InsertIntoRelation("ShoppingHistory", aLotOfRecords)
userDactor ! RelationQuery("ShoppingHistory")

def receive = {
  case akka.actor.Status.Failure(e) => retrySendingMessage(/* which one??*/)
}
```
_Remark:_ we may also define our own super-classes for those _unsafe_ message, so we could match them all together (like with `akka.actor.Status`)

Reply:
What do you think of renaming it to `SelectAllFromRelation`?
Because, what we provide here is like an SQL interface, right?

Reply:
Good reason for adding a specific `Failure` message. I will add this.

Can you elaborate on what you mean with *unsafe* messages?

`SelectAllFromRelation` sounds good to me as well, because it might enable us to add something like `SelectFromRelation(relationName: String, fields: Set[ColumnDef])` if we should ever need it.

Reply:
I thought of having different class hierarchies for messages between actors. For example like that:

| Root class | Childs | Description |
|:--|:--|:--|
| `akka.actor.Status` | `Success`, `Failure` | Messages from the `akka`-framework |
| `InternalMessageProtocol. InternalMessage` | `Ping`, `Pong`, `UpdateMapping`, `...` | Message used internally in the `adbms` framework. |
| `DefaultMessageProtocol. Status` | `InsertIntoRelation. Success\|Failure`, `SelectAllFromRelation. Success\|Failure` | _Unsafe_ messages, only used for testing, debugging and not meant for production. |
| `RequestResponseProtocol. Response` | `UserOp.Success\|Failure`, `...` | Production-grade messages defined by the developer (user of our framework). We only provide the base classes. |

We may rename `DefaultMessageProtocol` to something that sounds more badly.
**All names are just ideas and must be thought of more precisely to match the specific use cases.**

Reply:
Now I get it and I also think it makes a lot of sense. It's not really mission critical at the moment for our immediate tasks, right? If it is alright with you to defer this for now, I will make an issue out of this thought so we can return to it when we have time for the refactoring.

Reply:
Yeah, sure. It was just an anecdote to reason about my decision for having both `Failure` and `Success` defined there.